Parsers/ASimNetworkSession/ProductParsers/vimNetworkSessionMicrosoftLinuxSysmon.yaml

```diff
@@ -15,13 +15,61 @@ References:
 Description: |
   ASIM Sysmon for Linux Network Session Parser
 ParserName: vimNetworkSessionLinuxSysmon
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_ipv4_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstipaddr_has_any_ipv4_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dstportnumber
+    Type: int
+    Default: int(null)
+  - Name: url_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: httpuseragent_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: hostname_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: dvcaction
+    Type: dynamic
+    Default: dynamic([])
+  - Name: disabled
+    Type: bool
+    Default: false
 ParserQuery: |
-   let DirectionNetworkEvents =
-    Syslog 
-    | where SyslogMessage has_all ('<Provider Name="Linux-Sysmon"', '<EventID>3</EventID>')
-    | project-away ProcessName, ProcessID
-    | parse SyslogMessage with * '<Data Name="SourceIp">' SrcIpAddr:string '</Data>' *
-    | extend outbound = (SrcIpAddr == HostIP or SrcIpAddr in ('127.0.0.1', '0.0.0.0'))
+    let DirectionNetworkEvents =
+        Syslog | where not(disabled)
+        | where SyslogMessage has_all ('<Provider Name="Linux-Sysmon"', '<EventID>3</EventID>')
+        | project-away ProcessName, ProcessID
+        | parse SyslogMessage with * '<Data Name="SourceIp">' SrcIpAddr:string '</Data>' *
+        // ***************  Prefilterring *****************************************************************
+        | where (isnull(starttime) or TimeGenerated>=starttime) 
+            and (isnull(endtime) or TimeGenerated<=endtime) 
+            and (array_length(srcipaddr_has_any_ipv4_prefix)==0 
```

Comment:
You should match the both source and dest IP prefix lists to the Syslog Message in a single swoop at this stage:
- Less passes.
- Don't use the calculated field SrcIpAddr.
- And you anyways, rightfully, moved calculating it to afterwards.

Obviously this means adding post filtering for SrcIpAddr when you calculate it.